Time::HiRes::sleep: use (;$) prototype, like CORE::sleep

mauke opened the pull request:
Time::HiRes::sleep is supposed to be a drop-in replacement for CORE::sleep, but before this patch Time::HiRes::sleep had no prototype, leading to the following incompatibilities:
    
```perl
CORE::sleep(1, "foo", "bar");  # syntax error
Time::HiRes::sleep(1, "foo", "bar");  # OK, ignores all but 1st argument

my @t = 42;
CORE::sleep @t;  # sleeps for one second (number of elements in @t)
Time::HiRes::sleep @t;  # sleeps for 42 seconds (@t in list context)

CORE::sleep 1, next if $foo;
# if $foo, then sleep for one second and start next loop iteration

Time::HiRes::sleep 1, next if $foo;
# parses as: Time::HiRes::sleep(1, next) if $foo;
# if $foo, then start next loop iteration; no delay
```

Fixes #23628.

-------------------------------------------------------------------------------
<!--
Significant or noteworthy changes to Perl must be documented in pod/perldelta.pod.

Consider if the changes in this pull request are worthy of a perldelta
entry, then pick the appropriate line below and remove the others.
-->
* This set of changes requires a perldelta entry, and it is included.

mauke committed:
Time::HiRes::sleep: use (;$) prototype, like CORE::sleep

Time::HiRes::sleep is supposed to be a drop-in replacement for
CORE::sleep, but before this patch Time::HiRes::sleep had no prototype,
leading to the following incompatibilities:

    CORE::sleep(1, "foo", "bar");  # syntax error
    Time::HiRes::sleep(1, "foo", "bar");  # OK, ignores all but 1st argument

    my @t = 42;
    CORE::sleep @t;  # sleeps for one second (number of elements in @t)
    Time::HiRes::sleep @t;  # sleeps for 42 seconds (@t in list context)

    CORE::sleep 1, next if $foo;
    # if $foo, then sleep for one second and start next loop iteration

    Time::HiRes::sleep 1, next if $foo;
    # parses as: Time::HiRes::sleep(1, next) if $foo;
    # if $foo, then start next loop iteration; no delay

Fixes #23628.

mauke committed:
perldelta for Time::HiRes 1.9779 (sleep prototype)

xenu commented on dist/Time-HiRes/t/alarm.t:
```diff
@@ -11,9 +11,9 @@ use Config;
-if (open(XDEFINE, "<", "xdefine")) {
-    chomp($xdefine = <XDEFINE> || "");
-    close(XDEFINE);
+if (open(my $fh, "<", "xdefine")) {
+    chomp($xdefine = <$fh> || "");
+    close($fh);
```
This bareword removal (also in sleep.t) is unrelated to the prototype issue. I think it should be put in a separate commit.

mauke replied:
I've factored out some of the test changes.

mauke committed:
Time::HiRes tests: remove bareword filehandles and '1;'

Bareword filehandles have been obsolete since 2000. There is no point in
adding '1;' to scripts; only modules have return values.

Leont commented on dist/Time-HiRes/HiRes.xs:
```diff
@@ -1019,6 +1019,7 @@ nanosleep(nsec)
 
 NV
 sleep(...)
+PROTOTYPE: ;$
```
The `PROTOTYPE` should probably align with the other XS keywords

mauke replied:
I've followed the existing alignment of `PROTOTYPE:` in `utime` and `stat`.

Leont replied:
I see. In that case either way is fine I suppose.
